ibft/storage/store.go

```diff
@@ -155,85 +153,6 @@ func (i *ibftStorage) CleanAllInstances(msgID []byte) error {
 	return nil
 }
 
-// SaveLastChangeRoundMsg updates last change round message
-func (i *ibftStorage) SaveLastChangeRoundMsg(msg *specqbft.SignedMessage) error {
-	i.forkLock.RLock()
-	defer i.forkLock.RUnlock()
-
-	var signers [][]byte
-	for _, s := range msg.GetSigners() {
-		signers = append(signers, uInt64ToByteSlice(uint64(s)))
-	}
-	signedMsg, err := i.fork.EncodeSignedMsg(msg)
-	if err != nil {
-		return errors.Wrap(err, "could not encode signed message")
-	}
-	return i.save(signedMsg, lastChangeRoundKey, msg.Message.Identifier, signers...)
-}
-
-// GetLastChangeRoundMsg returns last known change round message
-func (i *ibftStorage) GetLastChangeRoundMsg(identifier []byte, signers ...spectypes.OperatorID) ([]*specqbft.SignedMessage, error) {
-	i.forkLock.RLock()
-	defer i.forkLock.RUnlock()
-
-	if len(signers) == 0 {
-		res, err := i.getAll(lastChangeRoundKey, identifier[:])
-
-		if err != nil {
-			return nil, err
-		}
-		return res, nil
-	}
-
-	var res []*specqbft.SignedMessage
-	for _, s := range signers {
-		msg, found, err := i.get(lastChangeRoundKey, identifier[:], uInt64ToByteSlice(uint64(s)))
-		if err != nil {
-			return res, err
-		}
-		if !found {
-			return res, nil
-		}
-		sm := new(specqbft.SignedMessage)
-		if err := sm.Decode(msg); err != nil {
-			return res, err
-		}
-		res = append(res, sm)
-	}
-	return res, nil
-}
-
-// CleanLastChangeRound cleans last change round message of some validator, should be called upon controller init
-func (i *ibftStorage) CleanLastChangeRound(identifier []byte) error {
-	i.forkLock.RLock()
-	defer i.forkLock.RUnlock()
-
-	prefix := i.prefix
-	prefix = append(prefix, identifier[:]...)
-	prefix = append(prefix, []byte(lastChangeRoundKey)...)
-	n, err := i.db.DeleteByPrefix(prefix)
-	if err != nil {
-		return errors.Wrap(err, "failed to remove last change round")
-	}
-	i.logger.Debug("removed last change round", zap.Int("count", n),
-		zap.String("identifier", hex.EncodeToString(identifier)))
-	return nil
-}
-
-func (i *ibftStorage) CleanAllChangeRound() error {
```

Comment:
We don't need to remove this as it should be run in the migration. We should take all shares, go through them and run this function for each share's identifier.

Reply:
@nkryuchkov Can we move it to the migration file it self? so we wont have leftovers in the code?

Reply:
We can keep this function instead to avoid redundant calls to read the shares, plus a call per identifier to clean.

Reply:
@amir-blox It's implemented wrong: It deletes by `append(i.prefix, []byte(lastChangeRoundKey)...)`, however, `SaveLastChangeRoundMsg` does `append(i.prefix, identifier, []byte(lastChangeRoundKey)...`. I don't see a quick way to fix this